src/TelemetryChannels/ServerTelemetryChannel/Shared/Implementation/TelemetryBuffer.cs

```diff
@@ -103,6 +128,17 @@ public virtual void Process(ITelemetry item)
 
             lock (this)
             {
+                if (this.transmissionBuffer.Count >= this.MaximumBacklogSize)
+                {
+                    if (!this.itemDroppedMessageLogged)
+                    {
+                        TelemetryChannelEventSource.Log.ItemDroppedAsMaximumUnsentBacklogSizeReached(this.MaximumBacklogSize);
+                        this.itemDroppedMessageLogged = true;
+                    }
+
+                    return;
+                }
```

Comment:
In my opinion we need a high priority bug to track that the server telemetry channel needs rewriting, at least partially, to make sure it doesn't fail to send or drop items at the buffer level.  This is not the correct place to drop items in the server telemetry channel, if we can't get items through the buffer, through the serializer, and out over the wire or saved onto disk then we have a big problem with this channel implementation.